lib/matplotlib/axis.py

```diff
@@ -1459,13 +1482,46 @@ def convert_units(self, x):
 
         if self.converter is None:
             return x
+
+        if hasattr(self.converter, 'convert'):
+            if hasattr(self.converter, 'to_numeric'):
+                to_numeric = getattr(
+                    self.converter, '_resolved_collision_to_numeric', None)
+
+                if to_numeric is None:
+                    # if convert and to_numeric are both defined, only use
+                    # to_numeric if it has been overriden
+                    for klass in self.converter.__class__.__mro__:
+                        if (klass is not munits.ConversionInterface
+                                and 'to_numeric' in klass.__dict__):
+                            to_numeric = self.converter.to_numeric
+                            break
+
+                    self.converter._resolved_collision_to_numeric = (
+                        to_numeric or self.converter.convert)
+
+            if to_numeric is None:
+                cbook.warn_deprecated(
+                    since="3.2", pending=True, obj_type='function',
+                    message=f"Converter {type(self.converter)} should "
+                    "implement 'to_numeric' rather than 'converter', and may "
+                    "need to override 'from_numeric'.")
+
+                to_numeric = self.converter.convert
+        else:
+            to_numeric = self.converter.to_numeric
+
         try:
-            ret = self.converter.convert(x, self.units, self)
+            ret = to_numeric(x, self.units, self)
         except Exception as e:
             raise munits.ConversionError('Failed to convert value(s) to axis '
                                          f'units: {x!r}') from e
         return ret
 
+    @cbook.deprecated("3.2", alternative="convert_to_numeric")
```

Comment:
3.2 is already out. This can only target 3.3.
```suggestion
    @cbook.deprecated("3.3", alternative="convert_to_numeric")
```

Reply:
done